ruby/tests/common_tests.rb

```diff
@@ -0,0 +1,1110 @@
+# Defines tests which are common between proto2 and proto3 syntax.
+#
+# Requires that the proto messages are exactly the same in proto2 and proto3 syntax
+# and that the including class should define a 'proto_module' method which returns
+# the enclosing module of the proto message classes.
+module CommonTests
+  def test_defaults
+    m = proto_module::TestMessage.new
```

Comment:
Is it possible to define proto_module::TestMessage in the begining so that you don't need to write so many proto_module::TestMessage?

Reply:
Do you mean as a constant? The problem with that is that `proto_module` is a method and for different classes where `CommonTests` gets included, it can have a different value. And hence, a constant won't work.

The alternative is to define a method like `test_message_class` or so and use that, but I like having `proto_module::TestMessage.new` over `test_message_class.new`.